### PR TITLE
`Icon` web doc improvements (HDS-2092)

### PR DESCRIPTION
**Pull request by KristinLBradley**
_**Note:** I attempted to improve organization & hierarchy of different text sections in the “Code” tab. Still not sure is the info related to animation and installing in Ember vs React is in the best spot so please let me know if you have any ideas for further improvements._

### :pushpin: Summary
If merged, this PR further improves web docs for Icons

**Web docs preview:** https://hds-website-git-hds-2092-icon-docs-improvements-hashicorp.vercel.app/icons/usage-guidelines?tab=code

### :hammer_and_wrench: Detailed description
* Added a max-width style for `@stretched={{true}} example`
* Reorganized and edited “Code” tab documentation to improve clarity

### :link: External links

Jira ticket: [HDS-2092](https://hashicorp.atlassian.net/browse/HDS-2092)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2092]: https://hashicorp.atlassian.net/browse/HDS-2092?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by didoo — APPROVED** [review]
For me, it's **so much better** now. Everything flows, and the logical order of the content makes more sense.

One only thing: maybe consider moving this paragraph
> Icons can be used in many ways. The package can be installed as an [Ember addon](https://hds-website-git-hds-2092-icon-docs-improvements-hashicorp.vercel.app/icons/usage-guidelines?tab=code#using-icons-in-ember-apps) for the convenience of using a component with strong defaults. It can also be [consumed in React applications](https://hds-website-git-hds-2092-icon-docs-improvements-hashicorp.vercel.app/icons/usage-guidelines?tab=code#using-icons-in-react-apps) via direct import of the SVG file or as a standalone React/SVG icon component.

as introduction, between "Installation" and "Using icons in Ember apps" and see how it feels for you.

**Review by alex-ju — COMMENTED** [review]
Since you're at it – could you set the default size to 16 in the 'Component API' section? I see we mention this in the docs, but not highlighting it in the API section

**Comment by KristinLBradley** [comment]
> Since you're at it – could you set the default size to 16 in the 'Component API' section? I see we mention this in the docs, but not highlighting it in the API section

Actually I have the default specified in the code but it's not displaying...
```
<C.Property @name="size" @type="number" @default="16" @values={{array 16 24}}>
  Sets the size of the icon in pixels. Only two sizes are supported. (Setting a non-standard size will cause the SVG to render at the specified size but it will be invisible.)
</C.Property>
```

**Comment by didoo** [comment]
> @values={{array 16 24}}

I suspect you have to say `@values={{array "16" "24"}}`. Can you try if this does the trick?

**Comment by KristinLBradley** [comment]
> > @values={{array 16 24}}
> 
> I suspect you have to say `@values={{array "16" "24"}}`. Can you try if this does the trick?

@didoo Thanks. I just realized that I had mismatched data times (string vs. number). I'll make them all strings.

File: website/docs/icons/usage-guidelines/partials/code/component-api.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <C.Property @name="color" @type="string" @default="currentColor">
     The `@color` argument can be used to change the color. It works by setting the value of the icon SVG’s `fill` property.
   </C.Property>
-  <C.Property @name="size" @type="number" @default="16" @values={{array 16 24}}>
+  <C.Property @name="size" @type="number" @default="16" @values={{array "16" "24"}}>
```
Comment (KristinLBradley) [inline website/docs/icons/usage-guidelines/partials/code/component-api.md:10]: @alex-ju @didoo Fixed the mismatched value types. Thanks for catching that.

**Review by jorytindall — APPROVED** [review]
:shipit:

**Pull request merged**